src/AppInstallerRepositoryCore/Rest/RestSource.cpp

```diff
@@ -0,0 +1,272 @@
+                {
+                    return GetLatestAvailableVersion();
+                }
+                else if (versionKey.Version.empty())
+                {
+                    for (const auto& versionInfo : m_package.versions)
+                    {
+                        if (CaseInsensitiveEquals(versionInfo.GetChannel().ToString(), versionKey.Channel))
+                        {
+                            packageVersion = std::make_shared<PackageVersion>(source, m_package.packageInfo, versionInfo);
+                            break;
+                        }
+                    }
+                }
+                else if (versionKey.Channel.empty())
+                {
+                    for (const auto& versionInfo : m_package.versions)
+                    {
+                        if (CaseInsensitiveEquals(versionInfo.GetVersion().ToString(), versionKey.Version))
+                        {
+                            packageVersion = std::make_shared<PackageVersion>(source, m_package.packageInfo, versionInfo);
+                            break;
+                        }
+                    }
+                }
+
+                return packageVersion;
+            }
+
+            bool IsUpdateAvailable() const override
+            {
+                return false;
+            }
+
+            bool IsSame(const IPackage* other) const override
+            {
+                const AvailablePackage* otherAvailablePackage = dynamic_cast<const AvailablePackage*>(other);
+
+                if (otherAvailablePackage)
+                {
+                    return Utility::CaseInsensitiveEquals(m_package.packageInfo.packageIdentifier, otherAvailablePackage->m_package.packageInfo.packageIdentifier);
+                }
+
+                return false;
+            }
+        };
+    }
+
+    RestSource::RestSource(const SourceDetails& details, std::string identifier, RestClient&& restClient)
+        : m_details(details), m_identifier(std::move(identifier)), m_restClient(std::move(restClient))
+    {
+    }
+
+    const SourceDetails& RestSource::GetDetails() const
+    {
+        return m_details;
+    }
+
+    const RestClient& RestSource::GetRestClient() const
+    {
+        return m_restClient;
+    }
+
+    const std::string& RestSource::GetIdentifier() const
+    {
+        return m_identifier;
+    }
+
+    SearchResult RestSource::Search(const SearchRequest& request) const
+    {
+        // Note: Basic search functionality to fetch everything.
+        RestClient::SearchResult results = m_restClient.Search(request);
+        SearchResult searchResult;
+
+        std::shared_ptr<const RestSource> sharedThis = shared_from_this();
+        for (auto& result : results.Matches)
+        {
+            std::unique_ptr<IPackage> package = std::make_unique<AvailablePackage>(sharedThis, std::move(result));
+            PackageMatchFilter packageFilter({}, {}, {});

```

Comment:
>packageFilter({}, {}, {}); [](start = 31, length = 26)

Are we going to fill this in the future?

Reply:
I dont think so..

---
In reply to: [594736746](https://github.com/microsoft/winget-cli/pull/765#discussion_r594736746) [](ancestors = 594736746)

Reply:
It's used in some of the sorting logic for getting the order of matches and display to user more relevant matches. We might want to add to todo for improve item in the future


---
In reply to: [594738139](https://github.com/microsoft/winget-cli/pull/765#discussion_r594738139) [](ancestors = 594738139,594736746)